pwiz_tools/Skyline/Model/Results/ChromatogramCache.cs

```diff
@@ -1230,6 +1231,104 @@ public IEnumerable<ChromKeyIndices> GetChromKeys(MsDataFileUri msDataFilePath)
             }
         }
 
+        private ChromatogramCache UpdateOptimizationSteps(SrmDocument doc)
+        {
+            if (doc?.MeasuredResults?.Chromatograms == null || Version >= CacheFormatVersion.Seventeen)

```

Comment:
Can "doc" ever actually be null here? I couldn't find a code path that might be passing null in here, but I might have missed it.

Reply:
I don't think it can be null, but better to check for it than throw an exception here?